packages/jellyfish-testing/src/testingwrapper.ts

```diff
@@ -0,0 +1,32 @@
+import { RegTestFoundationKeys } from '@defichain/jellyfish-network'
+import { ContainerGroup, DeFiDContainer, MasterNodeRegTestContainer } from '@defichain/testcontainers'
+import { Testing, TestingGroup } from '.'
+
+export class TestingWrapper {
+  create (): Testing
+  create (n: number): TestingGroup
+  create (n: number, init: (index: number) => DeFiDContainer): Testing | TestingGroup
+
+  create (n?: number, init?: (index: number) => DeFiDContainer): Testing | TestingGroup {
+    if (init === undefined) {
+      init = (index: number) => new MasterNodeRegTestContainer(RegTestFoundationKeys[index])
+    }
```

Comment:
I think we might be able to hoist this out of the function?

```ts
type InitDeFiContainerFn = (index: number) => DeFiDContainer

function defaultInitDeFiContainer (index: number) {
  return new MasterNodeRegTestContainer(RegTestFoundationKeys[index])
}
```

```ts
create (n?: number, init: InitDeFiContainerFn = defaultInitDeFiContainer): Testing | TestingGroup {
```

Reply:
thanks for this one too